### PR TITLE
libappindicator: remove python2

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
:wave: Would be nice if, in general, less things depended on this in nixpkgs.
stuff found when doing https://github.com/NixOS/nixpkgs/pull/74295#issuecomment-559217359

It's possible something needed this, but idc, let them break for now :rofl: .

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by Ma27 — APPROVED** [review]

File: pkgs/development/libraries/libappindicator/default.nix [hunk 1]
```diff
@@ -25,7 +24,7 @@ in stdenv.mkDerivation rec {
     sha256 = "17xlqd60v0zllrxp8bgq3k5a1jkj0svkqn8rzllcyjh8k0gpr46m";
   };
 
-  nativeBuildInputs = [ pkgconfig autoconf ];
+  nativeBuildInputs = [ pkgconfig autoconf vala ];
```
Comment (jtojnar) [inline pkgs/development/libraries/libappindicator/default.nix]: `gobject-introspection` not needed, or falling back to pre-generated GIRs from the tarball?

Comment (worldofpeace) [reply pkgs/development/libraries/libappindicator/default.nix]: ah, it's in buildInputs. will move.

**Review by jtojnar — APPROVED** [review]
Did not test but LGTM. We could probably get rid of `monoSupport` as well.

**Comment by jtojnar** [comment]
Nice, this reduced my system from 171 boxes to 148:

old

new

<details>
<summary>Drv tool</summary>

Based on <https://github.com/NixOS/nix/issues/1918#issuecomment-444110195>

```bash
#!/usr/bin/env nix-shell
#!nix-shell -p graphviz -p python3.pkgs.xdot -i bash

if (( $# != 2 )); then
    echo "usage: locate.sh <parent> <child>"
    exit 1
fi

if [[ "$1" =~ \.drv$ ]]; then
    parent=$1
else
    parent=$(nix-instantiate '<nixpkgs>' -A "$1")
fi

if [[ "$2" =~ \.drv$ ]]; then
    child=$2
else
    child=$(nix-instantiate '<nixpkgs>' -A "$2")
fi

nix-store -q --graph "$parent" | dijkstra -da "$child" | gvpr -c 'N[dist>1000.0]{delete(NULL, $)}' | xdot -

```

</details>

**Comment by nyanloutre** [comment]
GTK2 build of this library is failing since this commit: https://hydra.nixos.org/build/107472670/nixlog/1

**Comment by nyanloutre** [comment]
I made a fix here: #74733

**Pull request merged**